models/extensionRequests.js

```diff
@@ -0,0 +1,89 @@
+const firestore = require("../utils/firestore");
+const extensionRequestsModel = firestore.collection("extensionRequests");
+const { buildExtensionRequests, formatExtensionRequest } = require("../utils/extensionRequests");
+
+/**
+ * Create Extension Request
+ * @param extensionRequestData { Object }: Body of the extension request
+ */
+const createExtensionRequest = async (extensionRequestData) => {
+  try {
+    const request = {
+      timestamp: (new Date().getTime() / 1000).toFixed(0),
+      ...extensionRequestData,
+    };
+    return await extensionRequestsModel.add(request);
+  } catch (err) {
+    logger.error("Error in adding extension requests", err);
+    throw err;
+  }
+};
+
+/**
+ * Updates Extension Request
+ *
+ * @param extensionRequestData { Object }: extension request data object to be stored in DB
+ * @param extensionRequestId { string }: extensionRequestId which will be used to update the task in DB
+ * @return {extensionRequestResult : Object}
+ */
+const updateExtensionRequest = async (extensionRequestData, extensionRequestId) => {
+  try {
+    const extensionRequest = await extensionRequestsModel.doc(extensionRequestId).get();
+    await extensionRequestsModel.doc(extensionRequestId).set({
+      ...extensionRequest.data(),
+      ...extensionRequestData,
+    });
+    return extensionRequestId;
+  } catch (err) {
+    logger.error("Error in updating task", err);
+    throw err;
+  }
+};
+
+/**
+ * Fetch all Extension Requests
+ * @param extensionRequestquery { Object }: Body of the extension request
```

Comment:
Can we further document the extensionRequestquery param to make it clear what properties are required? 

[Link to JSDoc](https://jsdoc.app/tags-param.html)

Reply:
sure will do that